### PR TITLE
Support for MSC3882 revision 1

### PR DESCRIPTION
**Pull request by hughns**
Availability of MSC3882 is now via /capabilities not /versions.

This needs to be merged after https://github.com/matrix-org/matrix-js-sdk/pull/3228

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Support for MSC3882 revision 1 ([\#10443](https://github.com/matrix-org/matrix-react-sdk/pull/10443)). Contributed by @hughns.

### CODE REVIEW

File: src/components/views/settings/devices/LoginWithQRSection.tsx [hunk 1]
```diff
@@ -15,15 +15,21 @@ limitations under the License.
...
-import type { IServerVersions } from "matrix-js-sdk/src/matrix";
 import { _t } from "../../../../languageHandler";
 import AccessibleButton from "../../elements/AccessibleButton";
 import SettingsSubsection from "../shared/SettingsSubsection";
 
 interface IProps {
     onShowQr: () => void;
     versions?: IServerVersions;
+    // we can't use the capabilities type from the js-sdk because it isn't exported
+    capabilities?: Record<string, any>;
```
Comment (weeman1337) [inline src/components/views/settings/devices/LoginWithQRSection.tsx:32]: Is there anything to stop us from exporting it?

**Review by weeman1337 — APPROVED** [review]
Looks good to me. :+1: 

 See my comment about the export from JS SDK.

File: src/components/views/settings/devices/LoginWithQRSection.tsx (hunk 1 above)
Comment (weeman1337) [reply src/components/views/settings/devices/LoginWithQRSection.tsx:32]: > I can make another PR for this if you like?

:+1: Same name would just be an inconvenience by using an alias. But if creating a new export you can just rename it to something more specific.

Comment (hughns) [reply src/components/views/settings/devices/LoginWithQRSection.tsx:32]: I've created https://github.com/matrix-org/matrix-js-sdk/pull/3266 to do the export.

**Pull request merged**